### PR TITLE
Only allow `-File` to accept `.ps1` scripts on Windows

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

A change was made previously to support shebang on Linux to accept files without requiring .ps1 extension.  However, Windows doesn't support shebang, so we should be consistent with Windows PowerShell and only allow for scripts with .ps1 extension. 

Fix is to update consolehost `-File` parsing path to check if the file has a `.ps1` extension on Windows and return an error.
A bunch of tests had to be updated with the new behavior.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/7904
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs [hunk 1]
```diff
@@ -1159,6 +1159,16 @@ static object ConvertToBoolIfPossible(string arg)
                         showHelp: true);
                     return false;
                 }
+#if !UNIX
+                // Only do the .ps1 extension check on Windows since shebang is not supported
+                else if (!Path.GetExtension(_file).Equals(".ps1", StringComparison.OrdinalIgnoreCase))
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs:1164]: Really we have no need to do full path analysis for path extension: 
```suggestion
                else if (!_file.EndsWith(".ps1", StringComparison.OrdinalIgnoreCase))
```

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs [hunk 2]
```diff
@@ -1857,13 +1857,15 @@ private void DoRunspaceInitialization(bool skipProfiles, string initialCommand,
 
                 Pipeline tempPipeline = exec.CreatePipeline();
                 Command c;
+#if UNIX
                 // if file doesn't have .ps1 extension, we read the contents and treat it as a script to support shebang with no .ps1 extension usage
                 if (!Path.GetExtension(filePath).Equals(".ps1", StringComparison.OrdinalIgnoreCase))
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs:1862]: Old code but the same comment.

Comment (SteveL-MSFT) [reply src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs]: will update

File: test/xUnit/csharp/test_CommandLineParser.cs [hunk 3]
```diff
@@ -103,18 +103,27 @@ public static void TestDefaultParameterIsFileName_Not_Exist(params string[] comm
...
+                Assert.False(cpp.AbortStartup);
+                Assert.False(cpp.NoExit);
+                Assert.False(cpp.ShowShortHelp);
+                Assert.False(cpp.ShowBanner);
+                Assert.Equal(CommandLineParameterParser.NormalizeFilePath(tempPs1), cpp.File);
+                Assert.Null(cpp.ErrorMessage);
+            }
+            finally
+            {
+                File.Delete(tempPs1);
```
Comment (iSazonov) [inline test/xUnit/csharp/test_CommandLineParser.cs:125]: Should we remove tempFile too?

Comment (SteveL-MSFT) [reply test/xUnit/csharp/test_CommandLineParser.cs:125]: tempFile is renamed to tempPs1 so only tempPs1 actually exists

**Comment by SteveL-MSFT** [comment]
static analysis failures are not related to this PR, still not sure why those URLs fail in CI but work locally from my dev machine unless CI is using a proxy for internet.

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs [hunk 4]
```diff
@@ -1159,6 +1159,14 @@ static object ConvertToBoolIfPossible(string arg)
                         showHelp: true);
                     return false;
                 }
+#if !UNIX
+                // Only do the .ps1 extension check on Windows since shebang is not supported
+                else if (!_file.EndsWith(".ps1", StringComparison.OrdinalIgnoreCase))
```
Comment (PaulHigin) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs:1164]:
```suggestion
                if (!_file.EndsWith(".ps1", StringComparison.OrdinalIgnoreCase))
```

**Review by PaulHigin — CHANGES_REQUESTED** [review]

**Review by rjmholt — APPROVED** [review]

**Review by PaulHigin — APPROVED** [review]

**Comment by SteveL-MSFT** [comment]
Don't know why codeql is failing as it's complaining about `--print-diagnostics-summary` being an invalid switch, but it's clearly supported via https://codeql.github.com/docs/codeql-cli/manual/database-interpret-results/

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**